dev/dev_setup.sh

```diff
@@ -120,97 +167,140 @@ function install_ghcup {
 
 }
 
-function install_gmp {
-    if [ $CRYPTOL_PLATFORM = $MACOS12 ] || [ $CRYPTOL_PLATFORM = $MACOS14 ]; then
-        notice "Installing GMP via Homebrew, if it's not already installed"
-        logged brew install gmp
-        USED_BREW=true
-    else
-        notice "Did not install GMP. This script only supports macOS 14"
-    fi
+install_gmp() {
+    case $CRYPTOL_PLATFORM in
+        $MACOS12 | $MACOS14)
+            notice "Installing GMP via Homebrew, if it's not already installed"
+            logged brew install gmp && USED_BREW=true;;
+        $UBUNTU20 | $UBUNTU22)
+            notice "Installing GMP via apt-get, if it's not already installed"
+            logged apt-get install -y libgmp-dev libgmp10;;
+        *) unreachable "Unsupported platform; did not install GMP";;
+    esac
 }
 
-# Installs the two solvers required to run the test suite for the repo.
-# Users may want to install other solvers, and indeed the what4 solvers repo
-# includes a half dozen other solvers that are compatible with cryptol. 
-function install_solvers {
-
-    # Install the latest z3
-    if ! is_installed z3; then
-        notice "Installing z3 solver via Homebrew"
-        logged brew install z3
-        USED_BREW=true
-    fi
+install_zlib() {
+    case $CRYPTOL_PLATFORM in
+        $MACOS12 | $MACOS14)
+            notice "Installing zlib via Homebrew, if it's not already installed"
+            logged brew install zlib && USED_BREW=true;;
```

Comment:
note: I'm pretty sure this comes by default in all macs these days, but I added it here for completeness, on the off chance that somebody accidentally trashed their own version. Brew defaults to not linking it to anything to avoid borking with the system version.